keras/engine/training.py

```diff
@@ -129,6 +129,9 @@ class Model(base_layer.Layer, version_utils.ModelVersionSelector):
   model = tf.keras.Model(inputs=inputs, outputs=outputs)
   ```
 
+  Note: Only dicts, lists, and tuples of input tensors are supported. Nested
+  inputs are not supported (e.g. lists of list or dicts of dict).
```

Comment:
Is this a deliberate policy? Or can this be reconsidered in favour of supporting nested inputs?